keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -411,4 +417,29 @@ class ApplicationService(
         throw UnsupportedScmType(message = "Stash is currently the only supported SCM type")
     }
   }
+
+  private fun getUrl(version1: PublishedArtifact?, version2: PublishedArtifact?, artifact: DeliveryArtifact): String? {
+    return if (version1 != null && version2 != null) {
+      return if (artifact.sortingStrategy.comparator.compare(version1, version2) < 0) {
```

Comment:
There are a couple of utility functions you could use here in place of this code to make it more readable. Sorry I didn't think of that earlier: https://github.com/spinnaker/keel/blob/e6f6e6c7824a68164f32887d61d7800c2310c647/keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt#L267-L279

It would be nice to `ArtifactUtils`.

Reply:
I started doing it but it was more harm than good. I ended up using only one line, and it didn't make sense to me to export it via a bunch of files and classes..